### PR TITLE
feat: add types support to esm bundle for lightclient

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Make the lightclient more accessible.

**Description**

- Allow people to directly import the ESM bundle with full Types support 

**Steps to test or reproduce**

- Run tests

### CODE REVIEW

File: packages/types/src/test.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {Boolean, phase0} from "./sszTypes.js";
```
Comment (nflaig) [inline packages/types/src/test.ts]: What's the purpose of this file?

Comment (nazarhussain) [reply packages/types/src/test.ts]: Accidentaly committed.

**Review by wemeetagain — DISMISSED** [review]

File: packages/light-client/vite.config.ts [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
 
 import pkgJSON from "./package.json";
 
 export default mergeConfig(
   getBaseViteConfig(pkgJSON, {libName: "LightClient", entry: "src/index.ts"}),
   defineConfig({
+    plugins: [
+      dts({
+        rollupTypes: true,
+        bundledPackages: ["@lodestar/*", "@chainsafe/persistent-merkle-tree", "@chainsafe/bls", "@chainsafe/ssz"],
```
Comment (jeluard) [inline packages/light-client/vite.config.ts:15]: How do you come up with this list? Is there no way to automate this?

File: packages/light-client/package.json [hunk 3]
```diff
@@ -60,7 +60,7 @@
     "build:watch": "yarn run build --watch",
     "build:release": "yarn clean && yarn run build && yarn run build:dist",
     "check-build": "node -e \"(async function() { await import('./lib/index.js') })()\"",
-    "check-types": "tsc",
+    "check-types": "yarn run build:dist && tsc",
```
Comment (jeluard) [inline packages/light-client/package.json:63]: Why do we need this?

File: packages/light-client/vite.config.ts (hunk 2 above)
Comment (nazarhussain) [reply packages/light-client/vite.config.ts:15]: By building the types buyndle and investigating the types import in it. The ideal scenario we want is there should not be any thirdparty types import in types bundle.

File: packages/light-client/package.json (hunk 3 above)
Comment (nazarhussain) [reply packages/light-client/package.json:63]: This checks the tests file as well and one test file is using the types imported from the bundle. So we have to build it before running types check.

**Review by jeluard — APPROVED** [review]

File: packages/light-client/package.json (hunk 3 above)
Comment (nflaig) [reply packages/light-client/package.json:63]: Can we solve this differently? Having this build run as part of check-types is really annoying during local development. check-types is supposed to be fast and not create a bundle for production

This takes more time than all other check-types across all packages combined
```
@lodestar/light-client: vite v5.2.10 building for production...
@lodestar/light-client: transforming...
@lodestar/light-client: ✓ 562 modules transformed.
@lodestar/light-client: rendering chunks...
@lodestar/light-client: [vite:dts] Start generate declaration files...
@lodestar/light-client: computing gzip size...
@lodestar/light-client: [vite:dts] Start rollup declaration files...
@lodestar/light-client: Analysis will use the bundled TypeScript version 5.4.2
@lodestar/light-client: [vite:dts] Declaration files built in 6702ms.
@lodestar/light-client: dist/manifest.json            0.13 kB │ gzip:   0.11 kB
@lodestar/light-client: dist/lightclient.min.mjs  1,160.16 kB │ gzip: 341.51 kB │ map: 2,887.35 kB
@lodestar/light-client: ✓ built in 12.17s
```

It also creates a untracked file

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        packages/light-client/stats.html
```

Comment (nflaig) [reply packages/light-client/package.json:63]: it also randomly fails, see CI https://github.com/ChainSafe/lodestar/actions/runs/8911114728/job/24471848781#step:4:124

```
@lodestar/light-client: test/unit/webEsmBundle.browser.test.ts:3:63 - error TS2307: Cannot find module '../../dist/lightclient.min.mjs' or its corresponding type declarations.
@lodestar/light-client: 3 import {Lightclient, LightclientEvent, utils, transport} from "../../dist/lightclient.min.mjs";
```

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.19.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.19.0) :tada:

**Pull request merged**